### PR TITLE
Did the things you asked

### PR DESCRIPTION
**Pull request by AWalnut1**
Got the Dynamic Web Page thing up and running, about half way done, got a bit snagged on getting lists to work but im sure I'll be able to crack it on friday, also I'll be out tomorrow sorry.

### CODE REVIEW

**Review by gigamonkey — CHANGES_REQUESTED** [review]
Indeed, looks like you're on your way. this PR however has a ton of files from other projects in it. On Friday try to grab me so I can look at your codespace and see if we can clean this up a bit.

**Comment by AWalnut1** [comment]
Got lists working thank god. will actually write code tomorrow i swear

**Comment by AWalnut1** [comment]
got images working ahahahaha

**Comment by AWalnut1** [comment]
Almost done, got the links working put they are too large. 
also forgot to re request your review over the weekend sorry

File: projects/2023-01-17/public/scripts/script.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+const h1 = document.createElement ('a')
+
+h1.setAttribute('href', 'https://chat.openai.com/chat')
+
+h1.append (
+  document.createTextNode('ChatGPT')
+)
+
+c1.append (h1)
```
Comment (gigamonkey) [inline projects/2023-01-17/public/scripts/script.js:196]: If you inspect your page in the browser you'll see that you've created this structure:

2023-01-23 at 7 23 PM

Notice how the `a` element is not in the `p` element represented here by `c2` which it should be. Can you see why?

**Review by gigamonkey — CHANGES_REQUESTED** [review]
Really close! I think you can find the problem with the hint I gave you. But if not I can give you some more hints in class. There are also some details you could fill in if you look at the HTML in `index.html` and the html you generated, e.g. the `li` elements don't just contain plain text. And the `img` element has another attribute.

Beyond that, the other thing to do with this assignment is to try to write some functions to reduce the amount of duplicated code. Of which there is necessarily a lot. Basically just find a few lines of code that you do wrote over and over with maybe just different values and wrap that up in a function and use it instead.

**Comment by AWalnut1** [comment]
tried to make a helper function, didnt work out, commented it at the top of paragraph 2 so you can tell me how and why i messed it up. im just too tired right now to pursue this any further

**Review by gigamonkey — APPROVED** [review]
You hit all the main points. And I appreciate you trying to write a helper function, even if you didn't quite get it working in the time you had. You can merge this.

File: projects/2023-01-17/public/scripts/script.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+//const newParagraph = (text) = {
+//  let q = document.createTextElement('p')
+//
+//   q.append(
+//    document.createTextNode(text)
+//  )
+//
+//  body.append(q)
+//};
```
[the remark this answers is not in the document]
Comment (AWalnut1) [reply projects/2023-01-17/public/scripts/script.js:40]: ah

**Pull request merged**